pages/onboarding/index.tsx

```diff
@@ -1,15 +1,129 @@
+    bio: user.bio,
+    twitter_username: user.twitter_username,
+    linkedin_url: user.linkedin_url,
+    github_username: user.github_username,
+    website: user.website,
+  });
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isFormFilled, setIsFormFilled] = useState<boolean>(false);
+
+  const router = useRouter();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    setIsFormFilled(true);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    try {
+      setIsLoading(true);
+      await axios.patch(`/api/users/${user.id}`, formData);
+      setIsLoading(false);
+      router.push('/directory');
+    } catch (error) {
+      setIsLoading(false);
+      console.error(error);
+    }
+  };
+
+  if (!user) {
+    router.push('/');
+  }
+
+  const isSubmitDisabled = !isFormFilled && !isLoading;
 
-const Onboarding: NextPage = () => {
   return (
     <Container maxWidth="lg" className="flex justify-between pt-4">
       <Typography variant="h1" className="text-3xl">
         Onboarding
       </Typography>
+      <form onSubmit={handleSubmit}>
+        <TextField
+          label="Bio"
+          name="bio"
+          placeholder='Describe yourself!'
+          value={formData.bio}
+          onChange={handleChange}
+          multiline
+          rows={4}
+          fullWidth
+          inputProps={{ maxLength: 1000 }}
+        />
+        <TextField
+          label="Twitter Username"
+          placeholder='JonDoe24'
+          name="twitter_username"
+          value={formData.twitter_username}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <TextField
+          label="LinkedIn URL"
+          name="linkedin_url"
+          placeholder='https://www.linkedin.com/in/your-profile'
+          value={formData.linkedin_url}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <TextField
+          label="GitHub Username"
+          name="github_username"
+          placeholder='Jon Doe'
+          value={formData.github_username}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <TextField
+          label="Website"
+          name="website"
+          placeholder='https://www.your-website.com'
+          value={formData.website}
+          onChange={handleChange}
+          fullWidth
+          inputProps={{ maxLength: 200 }}
+        />
+        <Box mt={4} textAlign="right">
+          <Button variant="contained" color="primary" type="submit" disabled={isSubmitDisabled}>
+            {isLoading ? 'Submitting...' : 'Submit'}
+            Submit
+          </Button>
```

Comment:
The additional `Submit` here isn't needed , it causes `submitsubmit` to render

![image](https://user-images.githubusercontent.com/68509315/234377957-daf7d5e8-abfc-44cc-87ea-1f11e8eb31e1.png)